animate/interpolation.py

```diff
@@ -346,14 +346,12 @@ def clement_interpolant(source, target_space=None, boundary=False):
     P1 = firedrake.FunctionSpace(mesh, "CG", 1)
 
     # Determine target domain
-    if rank == 0:
-        tdomain = "{[i]: 0 <= i < t.dofs}"
-    elif rank == 1:
-        tdomain = f"{{[i, j]: 0 <= i < t.dofs and 0 <= j < {dim}}}"
-    else:
-        tdomain = (
-            f"{{[i, j, k]: 0 <= i < t.dofs and 0 <= j < {dim} and 0 <= k < {dim}}}"
-        )
+    tdomain_ranks = {
+        0: "{[i]: 0 <= i < t.dofs}",
+        1: f"{{[i, j]: 0 <= i < t.dofs and 0 <= j < {dim}}}",
+        2: f"{{[i, j, k]: 0 <= i < t.dofs and 0 <= j < {dim} and 0 <= k < {dim}}}",
+    }
+    tdomain = tdomain_ranks[rank]
```

Comment:
Do we need to keep this dictionary in memory? Would the linter not be okay with this?
```suggestion
    tdomain = {
        0: "{[i]: 0 <= i < t.dofs}",
        1: f"{{[i, j]: 0 <= i < t.dofs and 0 <= j < {dim}}}",
        2: f"{{[i, j, k]: 0 <= i < t.dofs and 0 <= j < {dim} and 0 <= k < {dim}}}",
    }[rank]
```

Reply:
Similarly below.

Reply:
Both done in 2946fc8a3241de5d1fa4732cda88d6a1ad16c0cf